### PR TITLE
chore: count brillig opcodes in nargo info

### PR DESCRIPTION
**Pull request by guipublic**
# Description

## Problem\*

Related to #5101

## Summary\*
Add the number of brillig opcodes in the info command

## Documentation\*

Check one:
- [X] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [X] I have tested the changes locally.
- [X] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by TomAFrench** [comment]
Can you break these out so that we can see individual functions? We could have a number of unconstrained functions but there's no way to see which of them is contributing to the number of opcodes.

**Comment by guipublic** [comment]
> Can you break these out so that we can see individual functions? We could have a number of unconstrained functions but there's no way to see which of them is contributing to the number of opcodes.

It's done

File: compiler/noirc_driver/src/program.rs [hunk 1]
```diff
@@ -29,4 +29,6 @@ pub struct CompiledProgram {
     pub warnings: Vec<SsaReport>,
     /// Names of the functions in the program. These are used for more informative debugging and benchmarking.
     pub names: Vec<String>,
+    /// Names of the unconstrained functions in the program.
+    pub brillig_names: Vec<String>,
```
Comment (jfecher) [inline compiler/noirc_driver/src/program.rs:33]: What is the ordering in this Vec? How do we know which name is which?

Comment (vezenovm) [reply compiler/noirc_driver/src/program.rs:33]: They are fetched in `compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs` in the same order that the brillig bytecode is collected. I just pushed a small update as well to improve this collection. We do a similar process into of `ssa.rs` for circuit names as we want to separate out the debug information from the actual bytecode artifact.

File: compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs [hunk 2]
```diff
@@ -824,6 +827,7 @@ impl<'a> Context<'a> {
                                 } else {
                                     let code =
                                         self.gen_brillig_for(func, arguments.clone(), brillig)?;
+
```
Comment (vezenovm) [inline compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs:830]:
```suggestion
```

**Review by jfecher — APPROVED** [review]

**Pull request merged**